docs/dev_guide/contents/extending_fido.rst

```diff
@@ -0,0 +1,501 @@
+* Process the response from your API into an instance of `.QueryResponseTable`.
+
+To process the query with the `.AttrWalker`, call the :meth:`.AttrWalker.create` method::
+
+  def search(self, query):
+    queries = walker.create(query)
+
+Assuming the walker is the one we defined above, queries would be a list of dicts with the attrs processed into query parameters for the API URL.
+
+.. note::
+
+   If you want your search method to be able to be called independently of Fido, then you should accept a variable number of positional arguments (``*args``) and they should have the AND operator applied to them.
+   This looks like::
+
+     def search(self, *args):
+         query = attr.and_(args)
+         queries = walker.create(query)
+
+
+Once the walker has processed the query into a form designed to be passed to your API, your ``search()`` method then needs to iterate over these parameters, make the requests and process the results in to a table.
+
+In the following example we pretend our client has a method ``_make_search(query_parameters)`` which takes the query parameters and makes a request to our API.
+We also pretend that the response is a json object in the form of a Python dictionary, which we want to put into the table.
+
+.. code-block:: python
+
+  def search(self, query):
+      queries = walker.create(query)
+
+      results = []
+      for query_parameters in queries:
+          results.append(self._make_search(query_parameters))
+
+      return QueryResponseTable(results, client=self)
+
+
+In reality, you probably want to post-process the results from your API before you put them in the table, they should be human readable first, with spaces and capitalisation as appropriate.
+
+
+The ``_can_handle_query`` method
+---------------------------------
+
+The next required method is ``_can_handle_query``, this method tells ``Fido`` if your client might be able to return results for a given query.
+If this method returns `True`, your clients ``search()`` method will be called for that query.
+This method gets passed each query (in its independent form), and must either return ``True`` or ``False``.
+
+A simple example, which just checks the type of ``attrs`` and not their values would be::
+
+  @classmethod
+  def _is_datasource_for(cls, *query):
+      query_attrs = set(type(x) for x in query)
+      supported_attrs = {a.Time, a.Level}
+      return supported_attrs.issuperset(query_attrs)
+
+Note, that this method is a class method, it gets called without instantiating your client to speed up the dispatching.
+
+
+Writing a Fetch Method
+----------------------
+
+The ``fetch()`` method of a Fido client is responsible for converting a set of search results (possibly sliced by the user) into a set of URLs to be downloaded.
+Due to the history of clients and how they were implemented in sunpy, some existing clients support use not through the ``Fido`` wrapper, this makes them appear more complex.
```

Comment:
"some existing clients support use not through" maybe "some existing clients support outside the Fido wrapper"?